### DISCUSSION TITLE
git push stalling

### DISCUSSION THREAD

**Discussion by nikkujensen** [sagemathinc/cocalc#6563] [Q&A]
I am trying to git push from my CoCalc project and it keeps asking me to punch in my password. Any idea why?
<img width="458" alt="image" src="https://user-images.githubusercontent.com/55917432/226298834-d4942d87-cbe6-4ff8-81c5-20fb6becf776.png">

**Comment by haraldschilly** [comment]
A passphrase is something you have to know, kind of a "password" to unlock the private key in that file. This is kind of out of scope, though. https://docs.github.com/en/authentication/connecting-to-github-with-ssh/working-with-ssh-key-passphrases

**Reply by mforbes** [reply]
@nikkujensen Can you push from your own computer? How are you connecting to CoCalc?  I usually connect via SSH, using the `ssh -A` option (`ForwardAgent yes` in `.ssh/config`), which then forwards my laptop's credentials to CoCalc, and then to github.  If you do this using the Web interface, I think you will need to authenticate at the start of each session:

```bash
ssh add
<enter your password for the key>
git push ...
```

Note that this password is for the `id_ed25519` key you generated and gave to github, not your CoCalc password.

**Reply by nikkujensen** [reply]
@haraldschilly yup, I got that. I have been able to `git push` before using my passphrase. One thing I changed recently is I used git lfs to upload some larger files. Then after using `git lfs track ...` I pushed changes and got this result. I have entered my passphrase several times then it just asks for it again

**Reply by haraldschilly** [reply]
@nikkujensen well, I don't know much about git-lfs. Maybe there is something of relevance going on with it? From the initial question of yours it sounded it is only plain git. In any case, it's possible to save the passphrase using the `ssh-agent`. i.e. https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent#adding-your-ssh-key-to-the-ssh-agent

Did you try that? That's the only idea I have – well – besides that there is a bug in git and/or git-lfs

**Reply by nikkujensen** [reply]
I tried this and it helped with getting the push to work (I skipped uploading a large file). Now I just need to figure out that other part.

Thanks!
